src/herder/HerderImpl.cpp

```diff
@@ -509,18 +509,46 @@ HerderImpl::processSCPQueue()
 void
 HerderImpl::processSCPQueueUpToIndex(uint64 slotIndex)
```

Comment:
It would be good to mention that `slotIndex` is an exclusive bound here; we will process messages up to but not including it. I might also prefer to rename it `endSlotIndex` or `slotIndexBound` (not `maxSlotIndex`, as that implies an inclusive range).

Reply:
I think this is inclusive as it's passed as `max` for `pop` - so I'll leave things as is

Reply:
Ah, yes you're right, I misread the existing `pop()`. _And_ I misread the docs for `std::upper_bound`, thinking the loop you wrote would be exclusive of `maxSlotIndex` too. Sigh. Sorry.